stock_barcodes/wizard/stock_barcodes_read_picking.py

```diff
@@ -0,0 +1,293 @@
+# Copyright 2019 Sergio Teruel <sergio.teruel@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+import logging
+from odoo import api, _, fields, models
+from odoo.fields import first
+from odoo.addons import decimal_precision as dp
+
+_logger = logging.getLogger(__name__)
+
+
+class WizStockBarcodesReadPicking(models.TransientModel):
+    _name = 'wiz.stock.barcodes.read.picking'
+    _inherit = 'wiz.stock.barcodes.read'
+    _description = 'Wizard to read barcode on picking'
+
+    picking_id = fields.Many2one(
+        comodel_name='stock.picking',
+        string='Picking',
+        readonly=True,
+    )
+    candidate_picking_ids = fields.One2many(
+        comodel_name='wiz.candidate.picking',
+        inverse_name='wiz_barcode_id',
+        string='Candidate pickings',
+        readonly=True,
+    )
+    picking_product_qty = fields.Float(
+        string='Picking quantities',
+        digits=dp.get_precision('Product Unit of Measure'),
+        readonly=True,
+    )
+    picking_type_code = fields.Selection([
+        ('incoming', 'Vendors'),
+        ('outgoing', 'Customers'),
+        ('internal', 'Internal'),
+    ], 'Type of Operation')
+
+    def name_get(self):
+        return [
+            (rec.id, '{} - {} - {}'.format(
+                _('Barcode reader'),
+                rec.picking_id.name, self.env.user.name)) for rec in self]
```

Comment:
If not `picking_id` (starting from picking type), it should be reflected in the initial name.